src/jit/lsra.cpp

```diff
@@ -5667,16 +5666,15 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
+#ifdef _TARGET_ARM_
+        if (isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
+        {
+            unassignPhysReg(availablePhysRegInterval, currentInterval->registerType);
+#else
         if (intervalToUnassign != nullptr)
         {
```

Comment:
Strangely, applying the above isAssigned() to other platforms causes a segmentation fault. So, I have written the same with before for other platforms.